### PR TITLE
zerotier: update to 1.6.1

### PR DESCRIPTION
**Pull request by mwarning**
Signed-off-by: Moritz Warning <moritzwarning@web.de>

Maintainer: me
Compile tested: ramips/mt7620, Nexx wt3020, OpenWrt master
Run tested: ZeroTier starts and connects to public network

### CODE REVIEW

**Comment by neheb** [comment]
Interesting ARM failure.

**Comment by mwarning** [comment]
I can only find `openwrt/build_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-7.5.0_musl_eabi/gcc-7.5.0/gcc/config/aarch64/arm_neon.h:vrbitq_u8 (uint8x16_t __a)`. Maybe some include path needs to be changed. There are multiple arm_neon.h files in the build_dir.

File: net/zerotier/patches/0006-mangix.patch [hunk 1]
```diff
@@ -0,0 +1,24 @@
+From 6026141ac7958661df9b4c3d09591b51c04f38c3 Mon Sep 17 00:00:00 2001
+From: Moritz Warning <moritzwarning@web.de>
+Date: Sat, 21 Nov 2020 18:15:47 +0100
+Subject: [PATCH] mangix
```
Comment (neheb) [inline net/zerotier/patches/0006-mangix.patch:4]: I'd replace with https://github.com/zerotier/ZeroTierOne/pull/1110

Comment (mwarning) [reply net/zerotier/patches/0006-mangix.patch]: done

**Comment by mwarning** [comment]
Compile failure on arc_arc700 is in iproute, not ZeroTier.
Compile failure on arm_cortex-a15_neon-vfpv4 is funny:
```
node/AES_armcrypto.cpp:29:6: error: 'vrbitq_u8' was not declared in this scope
  y = vrbitq_u8(y);
````

**Comment by neheb** [comment]
```
--- a/node/Constants.hpp
+++ b/node/Constants.hpp
@@ -109,7 +109,7 @@
 #include <immintrin.h>
 #endif
 
-#if (defined(__ARM_NEON) || defined(__ARM_NEON__) || defined(ZT_ARCH_ARM_HAS_NEON))
+#if (defined(__aarch64__) || defined(ZT_ARCH_ARM_HAS_NEON))
 #if (defined(__APPLE__) && !defined(__LP64__)) || (defined(__ANDROID__) && defined(__arm__))
 #ifdef ZT_ARCH_ARM_HAS_NEON
 #undef ZT_ARCH_ARM_HAS_NEON
```

This ARM code looks based on ARM64.

**Comment by mwarning** [comment]
Looks good now. Ready to be merged.
Platform arc_arc700 seems to be broken unrelated to ZT.

**Pull request merged**